### ISSUE TITLE
pip-compile fails on the requirements list without blank lines, and doesn't fail when some blank lines are added

### ISSUE THREAD

**Issue by yurivict** [jazzband/pip-tools#2171]
pip-compile fails on this reqs.in file:
apsw
atomicwrites
attrs
Cerberus
coverage
deepdiff
dill
disjoint-set
distro
docopt
flake8
Flask
Flask-Cors
GPy
GPyOpt
graphviz
inspyred
intervaltree
Jinja2
junit2html
junitparser
lazy-loader
matplotlib
metric-learn
mplcursors
networkx
Nuitka
numpy
packaging
pandas
prettytable
psutil
pycryptodome
pymongo
pyrfr
pytest
pytest-cov
pytest-dependency
pytest-mock
pytest-timeout
pytest-xdist
python-ulid
PyYAML
pyzmq
scikit-learn
scipy
setproctitle
sobol-seq
statsmodels
tabulate2
termcolor
wrapt
xgboost-cpu

But pip-compile doesn't fail on this reqs.in which is only different in blank lines:
apsw
atomicwrites

attrs
Cerberus
coverage

deepdiff
dill
disjoint-set
distro
docopt

flake8
Flask
Flask-Cors
GPy
GPyOpt
graphviz
inspyred
intervaltree
Jinja2
junit2html
junitparser
lazy-loader
matplotlib
metric-learn
mplcursors
networkx
Nuitka
numpy
packaging
pandas
prettytable
psutil
pycryptodome
pymongo
pyrfr
pytest
pytest-cov
pytest-dependency
pytest-mock
pytest-timeout
pytest-xdist
python-ulid
PyYAML
pyzmq
scikit-learn
scipy
setproctitle
sobol-seq
statsmodels
tabulate2
termcolor
wrapt
xgboost-cpu

--------------------------------------------

Python-3.11
Linux

**Comment by sirosen** [comment]
Sorry this sat for so long with no reply!

I've just tried to reproduce and was unable to do so on the latest `pip-tools` and `pip` versions.

There's insufficient information here for me to do any further work -- I need a description of the error and ideally some information about the pip and pip-tools versions used. If you're able to give that info, I'm happy to take another look, but for now I'm going to label as CNR and close.
